README.md

```diff
@@ -0,0 +1,17 @@
+# Service Access Layer
+
+Provides tools to generate from XML description language and database bindings
+for LSST commands, events and telemetry. Depends on ts\_openslice.
```

Comment:
Its dependencies are more complicated than that. This is a start; it may not be complete:

* Vortex OpenSplice, from ADLink.
* I don't fully understand the ts_opensplice dependency. I know Taigo's Docker environments have set $OPSL_URI to look for OpenSplice's configuration file in ts_opensplice, but I don't know if they do anything else with that package.
* ts_idl is required (though really only to generate IDL files for ts_salobj).

I would be inclined to list the dependencies somewhere else. I put them in my doc/revision_history.rst because they can change over time. Another option is to make a section for it below. I worry it may end up being too long for a README.

Note: I don't think the backslash in `ts\_opensplice` is necessary, because the underscore is in the middle of a word.

Reply:
Dependency list:

# Dependencies

* **ts_openslice** for ADLink OpenSliceDDS to generate C++, Python, Java and LabView interfaces
* **ts_idl** for IDL generation, needed for **ts_salobj**

next push to this PR will have this. Contrary to popular believe, there **isn't** any other dependency - I verified that by installing on my new desktop, but having some problems on pristine docker centos:latest image. You are right about in_middle, fixed that.

Reply:
Interesting. So ts_opensplice really does include the full OpenSplice libraries? That would be nice. The acid test for "needs" is, I think:
* Does make_idl_files.py work? This is a low bar since it may not need OpenSplice at all.
* Does make_salpy_libs.py work?
* Can libraries be built for other supported languages?
* Do the unit tests pass?

Reply:
I tested that on Docker image, changed wording a bit. Generating cpp, Python and Java interfaces works. I will leave it to somebody confirming PR to do unit tests - or please pass me a description how to run those and I will run those. I verified binding functionality.

Reply:
Did you change the env variables OSPL_HOME and OSPL_URI so it was indeed using the ts_opensplice package? The defaults point to locations in /opt.

The python unit tests can be run with `pytest -v` (the -v is optional but I prefer the output). They require the Test and Script SALPY libraries, e.g. built using:
```
make_idl_files.py Test Script
```
That would be useful information for this README file.

I'm confused. On my Docker image OPSL_HOME points to 
I don't know how to run Java unit tests.

Reply:
K-T pointed me to sections 3.2.4 and 3.2.5 of LSE-70 which have reasonable descriptions of the topic types. A pointer here could be useful.

But beware: the topic types are first mentioned in section 2 and the information there is completely useless.

Reply:
This link will probably work to LSE-70: https://ls.st/LSE-70
but it's password-protected so not necessarily much use for a README.

Reply:
> At risk of asking a stupid question: did you change the env variables OSPL_HOME and OSPL_URI so you were using the ts_opensplice package? The defaults point to locations in /opt.

Yes, that's solved by smart update to setup.env, which autodefault OSPL_HOME to ts_opensplice package assuming you checked it out. It's a bit smarter - it prints error message it that location doesn't exist, and don't finish setting up if that's a case.

Reply:
Great!

Reply:
> At risk of asking a stupid question: did you change the env variables OSPL_HOME and OSPL_URI so you were using the ts_opensplice package? The defaults point to locations in /opt.
> 
> The python unit tests can be run with `pytest -v` (the -v is optional but I prefer the output). They require the Test and Script SALPY libraries, e.g. built using command-line command:
> 
> ```
> make_idl_files.py Test Script
> ```
> 
> That would be useful information for this README file.
> 
> I don't know how to run Java unit tests.

This is above scope of proposed change. make_idl_files.py doesn't run on my setup, and I have no knowledge how to run Java unit test either. Let's get this PR done and I can focus on those in another task.